### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by hogthrob**
Improved INA219 response time
Removed monitor port
Sound and Screen Handling refactoring

A lot of changes, but no change in functionality.
Intention is to prepare for future UI and function additions (such as my WIP TIG Welding Support)

### CODE REVIEW

**Comment by thomastech** [comment]
Hi hogthrob,

I made some changes to your PR and uploaded them to your refactor repo.
https://github.com/hogthrob/Sparky/issues/1

After you test it out please resubmit the PR with the revised files.

- Thomas

**Comment by hogthrob** [comment]
Will take a couple of days to retest it. Should have looked at the FOB key triggered announcements in the first place.

**Comment by thomastech** [comment]
> Will take a couple of days to retest it.

I decided to revise some lazy code that I pasted into the updated misc.cpp. I'll upload the revision to your Refactor branch today.

Removing the monitor_port directive from platformio.ini is a bit inconvenient for me, but I can understand why you did it. So I'm looking for a workaround: Do you know of a directive that forces monitor_port to use the Windows COMx found by the auto-detected upload_port? Something that could do this should make us both happy.

- Thomas

**Comment by hogthrob** [comment]
For me leaving out the monitor port makes my Windows machine automatically detect upload and monitor port  as identical (which they are in case of ESP32 obviously). 
I thought this is the default by PlatformIO. 
BTW, I think it is a major inconvenience in PlatformIO's approach that this particular setting has to be done in a by its nature shared project file.

File: src/PulseWelder/src/speaker.cpp [hunk 1]
```diff
@@ -140,9 +140,10 @@ void Speaker::addDigitSounds(uint32_t val) {
             val /= 10; // shift to next digit
           }
           if (count == 0) { count = 1; digits[0] = 0; }
-          // special case for value == 0 
-
-          for (; count; count-- ) { AddNumberToSequence(digits[count]); }
+          // special case for value == 0
+          for (; count; count-- ) {
+            AddNumberToSequence(digits[count-1]);
```
Comment (hogthrob) [inline src/PulseWelder/src/speaker.cpp:145]: @thomastech: Once you mentioned an issue with the number announcement, I immediately thought I did this off-by-one mistake. Well spotted!

**Comment by thomastech** [comment]
> For me leaving out the monitor port makes my Windows machine automatically detect upload and monitor port as identical (which they are in case of ESP32 obviously). I thought this is the default by PlatformIO.

Well, you would think it was that easy. But apparently I live (and suffer) by Murphy's rules.

In my installation the upload_port is correctly auto-detected, but monitor_port defaults to the wrong COMx. For example, upload uses COM4, but the serial terminal auto-detect incorrectly chooses COM5. 

The workaround for me was to use the platformio.ino directive to set the correct port. With the latest PR I'm back to typing the COMx value after each new flash. That's fun and games for awhile, until it's not. 

- Thomas

File: src/PulseWelder/src/speaker.cpp (hunk 1 above)
Comment (thomastech) [reply src/PulseWelder/src/speaker.cpp:145]: We've all done that.

**Comment by hogthrob** [comment]
Looked at your changes, tried them, made a very minor change (basically to get back to the original behavior when playing the promo message). So from my point of view we are good to go. Regarding the platformio.ini / COM port issue. You could do the same I had to do when the monitor_port was in: Changing the platformio.ini but never commiting this particular change.

**Comment by thomastech** [comment]
> You could do the same I had to do when the monitor_port was in: Changing the platformio.ini but never commiting this particular change.

But your ini revision will be removing monitor_port when this PR is committed. I'm still trying to find a universally acceptable solution to restore the monitor_port.

**Comment by thomastech** [comment]
>  made a very minor change (basically to get back to the original behavior when playing the promo message).

The current behavior is to wait at least SPLASH_TIME (2.5s), but this time can be much longer due to bluetooth scanning. The newly proposed change will further increase the splash's time before showing the home screen, which is not necessary. So would prefer that this change be reverted.
- Thomas

**Comment by hogthrob** [comment]
Ok, removed last commit. Regarding the port thing: If the assumption is correct, that most users don't have the problem you have and the implicit detection works for them, it is a service for those. If you don't like this, just change it back.

**Comment by thomastech** [comment]
Thanks, I'll commit your updated PR today.

> Regarding the port thing: If the assumption is correct, that most users don't have the problem you have and the implicit detection works for them, it is a service for those. 

I agree, specifying the monitor_port in the public release is undesirable. Unfortunately my Platformio IDE is being affected by an oddball bug in the IDE's port auto-detect code. So for now I will revert my local branch's platformio.ini while I patiently wait for a remedy from the developers.

- Thomas

**Comment by thomastech** [comment]
Update: The Platformio developer has offered to include an enhancement patch for the Serial Monitor port issue that was causing me grief. The issue affects some workstation configurations with several generic USB ports.

It will be released in Platformio 4.1.1.  Details: https://github.com/platformio/platformio-core/issues/3349

- Tom

**Comment by hogthrob** [comment]
Thats great! So it was a fixable issue, good news.

**Pull request merged**